packages/web/src/components/track/PremiumTrackSection.tsx

```diff
@@ -143,49 +140,37 @@ const LockedPremiumTrackSection = ({
     setLockedContentModalVisibility
   ])
 
-  const handleSendTip = useCallback(() => {
-    if (account) {
-      dispatch(beginTip({ user: tippedUser, source, trackId }))
-    } else {
-      dispatch(pushRoute(SIGN_UP_PAGE))
-      dispatch(showRequiresAccountModal())
-    }
+  const handleSendTip = useAuthenticatedCallback(() => {
+    dispatch(beginTip({ user: tippedUser, source, trackId }))
 
     if (lockedContentModalVisibility) {
       setLockedContentModalVisibility(false)
     }
   }, [
     dispatch,
-    account,
     tippedUser,
     source,
     trackId,
     lockedContentModalVisibility,
     setLockedContentModalVisibility
   ])
 
-  const handleFollow = useCallback(() => {
-    if (account) {
-      if (isPremiumContentFollowGated(premiumConditions)) {
-        dispatch(
-          socialActions.followUser(
-            premiumConditions.follow_user_id,
-            followSource,
-            trackId
-          )
+  const handleFollow = useAuthenticatedCallback(() => {
+    if (isPremiumContentFollowGated(premiumConditions)) {
+      dispatch(
+        socialActions.followUser(
+          premiumConditions.follow_user_id,
+          followSource,
+          trackId
         )
-      }
-    } else {
-      dispatch(pushRoute(SIGN_UP_PAGE))
-      dispatch(showRequiresAccountModal())
+      )
+    }
 
-      if (lockedContentModalVisibility) {
-        setLockedContentModalVisibility(false)
-      }
+    if (lockedContentModalVisibility) {
```

Comment:
Previously this closed the locked content modal only if not signed in. Is this behavior change ok?

Reply:
think so - played around with it and it seems right. it's now the same behavior as sendTip as well